documentation/technical_writing_slides.md

```diff
@@ -0,0 +1,452 @@
+    - Not everything can be auto-generated.
+- **Skimmable**
+    - Headings: should be descriptive and concise
+    - Hyperlinks: descriptive, not [click here](https://www.smashingmagazine.com/2012/06/links-should-never-say-click-here/)
+    - Use listings etc. where appropriate, no *"wall of text"*
+- **Exemplary** (some, but not too many)
+- **Consistent** (language, formatting, wording)
+- **Current**
+
+---
+
+## Be Clear About Your Audience (1/2)
+
+**User vs. developer documentation**:
+
+- Distinction important for larger projects, not so important for smaller projects
+- Sometimes also third category *"maintainer"*
+- User docs: How does the software work?
+- Dev docs: Why, not only how
+
+---
+
+From [py-RSE](https://merely-useful.tech/py-rse/documentation.html):
+
+> The best function names in the world aren't going to answer the questions “Why does the software do this?” and “Why doesn’t it do this in a simpler way?”
+
+---
+
+## Be Clear About Your Audience (2/2)
+
+[py-RSE](https://merely-useful.tech/py-rse/documentation.html) distinguishes novices, competent practitioners, and experts (following [Wilson 2019, Teaching Tech Together](https://doi.org/10.1201/9780429330704))
+
+- A **novice** doesn’t yet have a mental model of the domain. -> Needs **tutorials** that introduce key ideas one by one.
+    - Good example: [deal.ii tutorials](https://dealii.org/developer/doxygen/deal.II/Tutorial.html)).
+- A **competent practitioner** knows enough to accomplish routine tasks with routine effort. -> Needs **reference guides, cookbooks, and Q&A sites**.
+    - Good example: [preCICE config reference](https://precice.org/configuration-xml-reference.html)).
+- An **expert** need this material as well – nobody's memory is perfect.
+
+---
+
+## Style Guides (1/4)
+
+Consistent tone and style ...
+
+- makes content easier to read.
+- reduces the reader's cognitive load.
+- increases confidence in authority of content.
+
+Read more: [Write the Docs on style guides](https://www.writethedocs.org/guide/writing/style-guides/)
+
+---
+
+## Style Guides (2/4)
+
+- Different people write differently. Even the same person writes differently every day.
+- Style guide contains a set of standards.
+- Writer does not have to worry -> saves time.
+- Traditional style guides have been around as long as publishing, e.g. [The Chicago Manual of Style](https://www.chicagomanualofstyle.org/).
+
+---
+
+## Style Guides (3/4)
+
+Example: brief excerpt of [Google developer documentation style guide](https://developers.google.com/style)
+
+- **Timeless documentation**: Avoid words such as *"currently"*, *"in the future"*, *"soon"*
+- Use present tense, avoid *"will"*
+    - Bad: *Send a query to the service. The server will send an acknowledgment.*
+
+- **Exclamation points**: Don't use them. Rather use notices such as Note or Caution.
+
+---
+
+## Style Guides (4/4)
+
+- **Headings**: Use "Sentence case of Headings", not "Title Case for Headings"
```

Comment:
So we are doing it wrong in our slides? :scream:

Reply:
Only if we follow the Google style guide, which we don't.